va/va.go

```diff
@@ -455,7 +460,7 @@ func (va *ValidationAuthorityImpl) validate(
 	return validationRecords, nil
 }
 
-func (va *ValidationAuthorityImpl) validateChallenge(ctx context.Context, identifier identifier.ACMEIdentifier, challenge core.Challenge) ([]core.ValidationRecord, error) {
+func (va *ValidationAuthorityImpl) validateChallenge(ctx context.Context, identifier identifier.ACMEIdentifier, challenge core.Challenge, scope core.AuthorizationScope, regid int64) ([]core.ValidationRecord, error) {
```

Comment:
if one is willing to expend and make change to GRPC I think it'd better add regid to core.challenge too: it can hydrated from auth object in sa/model.go

Reply:
Done (`scope`).

The `regid` could get bundled in the Challenge with the current method but this won't be necessary if the `kid` is passed through from the WFE.

Reply:
WRT the Challenge.Scope; should this be removed in wfe.prepChallengeForDisplay?

Reply:
Oh right, `regid` is still needed. Makes sense to add it as well.